catalog/app/components/JsonEditor/State.js

```diff
@@ -142,9 +165,38 @@ function collectErrors(allErrors, itemAddress, value) {
   return errors
 }
 
+function doesPlaceholderPathMatch(placeholder, path) {
+  if (placeholder.length !== path.length) return false
+  return placeholder.every(
+    (item, index) => item === path[index] || item === JSON_POINTER_PLACEHOLDER,
+  )
+}
+
+// TODO: extend getJsonDictValue
+// TODO: return address too
+// NOTE: returns jsonDictItem by path from JsonDict handling placeholder paths too
+export function getJsonDictItemRecursively(jsonDict, parentPath, key) {
+  const addressPath = getAddressPath(typeof key === 'undefined' ? '' : key, parentPath)
+  const itemAddress = serializeAddress(addressPath)
+  const item = jsonDict[itemAddress]
+  if (item) return item
+
+  let weight = 0
+  let placeholderItem = undefined
+  Object.entries(jsonDict).forEach(([path, value]) => {
+    if (doesPlaceholderPathMatch(deserializeAddress(path), addressPath)) {
+      if (weight < addressPath.length) {
+        weight = addressPath.length
+        placeholderItem = value
+      }
+    }
+  }, {})
```

Comment:
the last argument (`thisArgs`) doesn't seem necessary here

Reply:
It's a leftover `.reduce` instead of `.forEach` implementation, thanks